content/authors/client-side-validation-with-react-alert/index.md

```diff
@@ -0,0 +1,366 @@
+Having a way to check if the data that a user is sending to the webserver of a web application is correct or valid is a feat that JavaScript enables on the web.
+
+This feature gives frontend developers the ability to create logic that checks the data that users type into the input fields of web apps.
+
+In this article, we’re going to have a look at how to do this by using an npm package called [react-alert](package link). react-alert makes it easier to render the error messages from the logic that developers create.
+
+You should have an idea of the following before you read this article any further so that there wouldn’t be any form of confusion.
+
+1. We’d be needing [NodeJs](https://nodejs.org) since we’re making use of an npm package called react-alert
+
+2. An idea of how conditional statements works in react/JavaScript
+
+3. How npm works and how to install npm packages as dependencies in your projects
+
+4. A little understanding of DOM Manipulation
+
+### Overview
+
+Before React, vanillaJS was the “go-to” solution for client-side validation. One would need to understand the fundamentals of DOM manipulation and event handling.
+Since the aim of this validation process is to check if the data or information sent by the user to the server is appropriate by listening to the events that take place on the webpage.
+
+These events range from the click of a button `onClick`, the mouse hovering effect on a particular element on the page `onMouseOver`, the submission of details entered into a webform via the `onSubmit` event and many more that I can’t mention. If you want to know more about events in JavaScript, you should check them out [here](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/Events).
+
+Performing these tasks (DOM manipulation) in vanillaJS isn’t a big deal because the DOM is available. But in React, the DOM isn’t available a virtual DOM is available instead.
+
+Although, manipulating the DOM in React undergoes the same process of doing that in vanillaJS but there are some quirks to it. One of them is how the `document` object isn’t available in React since it is being bundled with nodejs which doesn’t run on the client-side of a web browser.
+
```

Comment:
> Although manipulating the DOM in React is the same process as that of vanilla JS, there are some quirks to it. One is that the `Document` object is not available in React. This is because it is being bundled with Node.js which doesn’t run on the client-side of a web browser.

I feel this section should be: 

Although manipulating the DOM in React **works the same way in** vanilla JS, there are some quirks to it. One is that the `Document` object is not available in React. This is because it is being bundled with Node.js which doesn’t run on the client-side of a web browser.

Reply:
I've resolved thisat my end